src/commands/transaction/mosaicglobalrestriction.ts

```diff
@@ -0,0 +1,123 @@
+import { MosaicIdAliasValidator } from '../../validators/mosaicId.validator';
+import { NumericStringValidator } from '../../validators/numericString.validator';
+import { MosaicRestrictionKeyValidator } from '../../validators/restrictionKey.validator';
+import { MosaicRestrictionTypeValidator } from '../../validators/restrictionType.validator';
+
+export class CommandOptions extends AnnounceTransactionsOptions {
+    @option({
+        flag: 'm',
+        description: 'Mosaic identifier or @alias being restricted.',
+        validator: new MosaicIdAliasValidator(),
+    })
+    mosaicId: string;
+
+    @option({
+        flag: 'r',
+        default: '0',
+        description: '(Optional) Identifier of the mosaic providing the restriction key.',
+    })
+    referenceMosaicId: string;
+
+    @option({
+        flag: 'k',
+        description: 'Restriction key relative to the reference mosaic identifier.',
+        validator: new MosaicRestrictionKeyValidator(),
+    })
+    restrictionKey: string;
+
+    @option({
+        flag: 'v',
+        default: '0',
+        description: '(Optional) Previous restriction value.',
+        validator: new NumericStringValidator(),
+    })
+    previousRestrictionValue: string;
+
+    @option({
+        flag: 't',
+        default: 'NONE',
+        description: '(Optional) Previous restriction type. (NONE: no restriction, EQ: equal, NE: not equal, LT: less than,' +
+            'LE: less than or equal, GT: greater than, GE: greater than or equal)',
+        validator: new MosaicRestrictionTypeValidator(),
+    })
+    previousRestrictionType: string;
+
+    @option({
+        flag: 'V',
+        description: 'New restriction value.',
+        validator: new NumericStringValidator(),
+    })
+    newRestrictionValue: string;
+
+    @option({
+        flag: 'T',
+        description: 'New restriction type.',
+        validator: new MosaicRestrictionTypeValidator(),
+    })
+    newRestrictionType: string;
+}
+
+@command({
+    description: 'Set a global restriction to a mosaic',
+})
+export default class extends AnnounceTransactionsCommand {
+    constructor() {
+        super();
+    }
+    @metadata
+    execute(options: CommandOptions) {
+        const profile = this.getProfile(options);
+        const account = profile.decrypt(options);
+        const mosaicId = new MosaicIdAliasResolver().resolve(options);
+        const newRestrictionType = new RestrictionTypeResolver().resolve(options);
+        const restrictionKey = new RestrictionKeyResolver().resolve(options);
+        const newRestrictionValue = new RestrictionValueResolver().resolve(options);
+        const maxFee = new MaxFeeResolver().resolve(options);
+
+        const previousRestrictionType = Number(MosaicRestrictionType[options.previousRestrictionType as any]) as MosaicRestrictionType;
+        const previousRestrictionValue = UInt64.fromNumericString(options.previousRestrictionValue);
```

Comment:
Could we add this logic inside a resolver?

Reply:
This is an optional parameter, so I think we shouldn't add this inside a resolver.

Reply:
I guess we could create a resolver while maintaining the parameter optional by not asking the user for the parameter. Just to reuse the logic in both commands + validation, and have tests.